src/core/stream/period/period_stream.ts

```diff
@@ -120,18 +124,29 @@ export default function PeriodStream({
   // `null` when no Adaptation is chosen (e.g. no subtitles)
   const adaptation$ = new ReplaySubject<Adaptation|null>(1);
   return adaptation$.pipe(
-    switchMap((adaptation) => {
-      if (adaptation === null) {
+    switchMap((adaptation, switchNb) => {
+      /**
+       * If this is not the first Adaptation choice, we might want to apply a
+       * delta to the current position so we can re-play back some media in the
+       * new Adaptation to give some context back.
+       * This value contains this relative position, in seconds.
+       * @see reloadAfterSwitch
+       */
+      const relativePosAfterSwitch =
```

Comment:
Don't you think that using 3 ternary operators here is not easily readable?

Reply:
I see it becoming more and more common and find it readable if we know how to read it.

Basically:
```
const toto = conditionA ? resultA :
             conditionB ? resultB :
             conditionC ? resultC : 
                          resultD;
```

is the equivalent to:
```
let toto;
if (conditionA) {
  toto = resultA;
} else if (conditionB) {
  toto = resultB;
} else if (conditionC) {
  toto = resultC;
} else {
  toto = resultD;
}
```